http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,261 @@
+        HttpClientOptions options = new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(options).build();
+        httpClient.request(HttpMethod.GET, app.getPort(Protocol.HTTPS), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> request.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            String contentTypeOptionsHeader = httpClientResponse.getHeader("X-Content-Type-Options");
+                            assertNotNull(contentTypeOptionsHeader);
+                            assertEquals("nosniff", contentTypeOptionsHeader);
+                            assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                            checkpoint.flag();
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> {
+                    assertThat("Body response", body.toString().contains(GREETING));
+                    checkpoint.flag();
+                })
+                .onFailure(vertxTestContext::failNow);
+    }
+
+    @Test
+    @Order(6)
+    @DisplayName("Test sending and receiving custom frames over HTTP/2")
+    void customFramesTest(VertxTestContext context) {
+        AtomicLong timerId = new AtomicLong();
+        Checkpoint entry = context.checkpoint(2);
+        HttpClientOptions options = new HttpClientOptions().setSsl(true).setUseAlpn(true).setProtocolVersion(HttpVersion.HTTP_2)
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setVerifyHost(false);
+        httpClient = vertx.httpClientBuilder().with(options).build();
+        httpClient.request(HttpMethod.GET, app.getPort(Protocol.HTTPS), baseUri.getHost(), "/ping")
+                .onFailure(context::failNow)
+                .flatMap(request -> request.sendHead().map(head -> request))
+                .map(request -> {
+                    Future<HttpClientResponse> pong = request.response().onSuccess(response -> {
+                        response.customFrameHandler(frame -> {
+                            context.verify(() -> {
+                                assertEquals(PING_DATA, frame.payload().toString().toLowerCase());
+                                entry.flag();
+                            });
+                        });
+                    });
+                    request.exceptionHandler(context::failNow);
+                    vertx.setPeriodic(1000, timerID -> {
+                        timerId.set(timerID);
+                        request.writeCustomFrame(10, 0, Buffer.buffer("ping"));
```

Comment:
I have hard time to understand this. It goes to me, but it is really complex test, could you document that test in a code, please? I doubt most of us know how event.connection().ping etc works.

Reply:
just hints will do

Reply:
I've changed it the complex logic to an easier with `CompletableFuture`, let me know if you are ok with.